src/CBT.NuGet/Tasks/AggregatePackages.cs

```diff
@@ -0,0 +1,115 @@
+﻿using CBT.NuGet.Internal;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Collections.Generic;
+using System.IO;
+
+namespace CBT.NuGet.Tasks
+{
+    /// <summary>
+    /// Generate NuGet properties.
+    ///
+    /// Generate properties that contain the path and version of a given nuget package.
+    /// </summary>
+    public sealed class AggregatePackages : Task
+    {
+        /// <summary>
+        /// Gets or sets the msbuild item of that contains the packages to aggregate.
+        /// Example Input:
+        /// <PropertyGroup>
+        /// <!-- MSBuild property to set/override. -->
+        /// * adds the files to the aggregate
+        /// | removes the files from the aggregate
+        ///   <CBTAggregatePackage>NugetPath_LSBuild_Corext=$(NugetPath_LSBuild_Corext)*$(NugetPath_Microsoft_LSBuild_Extensions_SQLIS)|$(NugetPath_Microsoft_LSBuild_Excluded_Extensions_SQLIS);MyAggPkg=$(NugetPath_Azure_Corext)*$(NugetPath_Azure_Corext_AGG)</CBTAggregatePackage>
+        /// </PropertyGroup>
+        /// </summary>
+        [Required]
+        public string PackagesToAggregate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the full path of the props file that is written to.
+        /// </summary>
+        [Required]
+        public string PropsFile { get; set; }
+
+        /// <summary>
+        /// Gets or sets the root path of the packages to be aggregated.
+        /// </summary>
+        [Required]
+        public string AggregateDestRoot { get; set; }
+
+        public override bool Execute()
+        {
+            BuildEngine = new CBTBuildEngine();
```

Comment:
You shouldn't need to create a new `CBTBuildEngine` in this case since this code is running when the task is running as a real task.  Set this on line 91 instead